client/src/components/CustomSearchBar.js

```diff
@@ -8,19 +8,49 @@ import SearchIcon from "@material-ui/icons/Search";
 import LoginFormStyler from '../utils/LoginFormStyler';
 import axios from 'axios';
 import {useGlobalState} from '../context/GlobalState';
+import ReviewCard from './ReviewCard';
 
 const CustomSearchBar = (props) => {
+  const [globalState, globalSetState] = useGlobalState();
   const [state, setState] = useState({ searchVal: ''});
   const {classes} = props;
   
   const handleSubmit = async (event) => {
-    await axios.get(`${window.location.origin.toString()}/review/searchByReviewText`, {
+    if (state.searchVal === ''){
+      await axios.get(`${window.location.origin.toString()}/review/getAllReviews`, {})
+      .then(res => {
+          let allReviews = [];
+          res.data.data.forEach(x => allReviews.push(
+              <ReviewCard
+                  key={x['_id']}
+                  heading={`${x['User']['UserLogin']}, ${(new Date(x['ReviwedAt'])).toLocaleString('default', { month: 'short', day: 'numeric', year: 'numeric'})} at ${(new Date(x['ReviwedAt'])).toLocaleString('default', { timeStyle: 'long'})}`} 
+                  brandName={x['Product']['ProductBrand']} 
+                  productName={x['Product']['ProductName']} 
+                  review={x['ReviewText']}
+              />));
+          setState({...state, allReviewCards: allReviews, storedReviewCards: allReviews});
```

Comment:
Why do we need to store it twice in both allReviewCards and storedReviewCards ?

Reply:
can u get on a call thx

Reply:
removed double storage